### PR TITLE
feat: noco installaion script

### PR DESCRIPTION
**Pull request by DarkPhoenix2704**
## Change Summary

Add Minio to installation script

- [x] Minio with local host

## Change type

- [x] feat: (new feature for the user, not a new feature for build script)

### CODE REVIEW

File: docker-compose/setup-script/noco.sh [hunk 1]
```diff
@@ -574,64 +665,44 @@ services:
...
-    volumes:
-      - ./nginx:/etc/nginx/conf.d
+  traefik:
+    image: traefik:v3.1
+    command:
+      - "--api.insecure=true"
+      - "--providers.docker=true"
+      - "--entrypoints.web.address=:80"
+      - "--providers.docker.exposedByDefault=false"
+      - "--entrypoints.minio.address=:9000"
```
Comment (pranavxc) [inline docker-compose/setup-script/noco.sh:675]: Hope only if minio enabled we need to set this

File: docker-compose/setup-script/noco.sh [hunk 2]
```diff
@@ -803,28 +839,11 @@ message_arr+=("Update script: update.sh")
...
-
-  echo "Restarting nginx to apply the new certificates"
-  # Reload nginx to apply the new certificates
-  $DOCKER_COMMAND compose exec nginx nginx -s reload
-
-  message_arr+=("NocoDB is now available at https://$DOMAIN_NAME")
-
-elif [ -n "$DOMAIN_NAME" ]; then
+if [ -n "$DOMAIN_NAME" ]; then
   message_arr+=("NocoDB is now available at http://$DOMAIN_NAME")
```
Comment (pranavxc) [inline docker-compose/setup-script/noco.sh:847]: If SSL enabled let's include `https`

**Comment by pranavxc** [comment]
Screenshot 2024-08-21 at 10 55 09 PM

 Saw above error while testing in cloud.

**Comment by pranavxc** [comment]
Somehow certificate generation fails for `noco.nocodb.dev`, is anything missing here?

Screenshot 2024-08-22 at 11 32 08 AM

Screenshot 2024-08-22 at 11 34 18 AM

Screenshot 2024-08-22 at 11 27 42 AM

**Comment by pranavxc** [comment]
With SSL option and domain name everything works as expected. Without SSL option it still redirecting to `https`, tried it in cognito window as well. 

Screenshot 2024-08-22 at 4 40 25 PM
Screenshot 2024-08-22 at 4 40 46 PM
Screenshot 2024-08-22 at 4 44 53 PM


Following may help : https://community.traefik.io/t/how-to-disable-http-to-https-redirect/18325

File: docker-compose/setup-script/noco.sh [hunk 3]
```diff
@@ -643,17 +611,58 @@ cat <<EOF >> docker-compose.yml
...
+      - ./minio:/export
+    command: -c 'mkdir -p /export/nocodb && /usr/bin/minio server /export'
+    labels:
+      - "traefik.enable=true"
+      - "traefik.http.services.minio.loadbalancer.server.port=9000"
+      - "traefik.http.routers.minio.rule=Host(\`${CONFIG_MINIO_DOMAIN_NAME}\`)"
+EOF
+# If minio SSL is enabled, set the entry point to websecure
 fi
+    if [ "${CONFIG_SSL_ENABLED}" = "Y" ]; then
```
Comment (pranavxc) [inline docker-compose/setup-script/noco.sh:641]: I guess we have to check `CONFIG_MINIO_SSL_ENABLED` here

Comment (DarkPhoenix2704) [reply docker-compose/setup-script/noco.sh]: updated

**Review by pranavxc — APPROVED** [review]

**Pull request merged**